### PR TITLE
Feature/add logging

### PR DESCRIPTION
**Pull request by mikeSmith08**
- Add in logging of the request id and session id
 - Made sure that the Person and Address on the request are validated
 - Make sure that request id and session id are passed to the smart stub.

### CODE REVIEW

**Review by dwybourn — DISMISSED** [review]

File: api/src/main/resources/application.yml [hunk 1]
```diff
@@ -35,4 +35,4 @@ logging:
     uk.gov.dhsc: ${vcap.services.variable-service.credentials.dwp-api-app-loglevel:debug}
 
 dwp:
-  uri: ${vcap.services.htbhf-dwp-api.credentials.uri:http://localhost:8080}
+  uri: ${vcap.services.htbhf-dwp-api.credentials.uri:http://localhost:8091/v1/dwp/benefits}
```
Comment (YetAnotherMatt) [inline api/src/main/resources/application.yml]: according to our new ports spec, the smart stub should be 8120

Comment (YetAnotherMatt) [reply api/src/main/resources/application.yml]: https://departmentofhealth-htbhf.github.io/PORTS.html

File: api/src/main/java/uk/gov/dhsc/htbhf/dwp/requestcontext/RequestIdFilter.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    /**
+     * The name of the session id header.
+     */
+    public static final String SESSION_ID_HEADER = "X-SESSION-ID";
+
+    private final RequestContext requestContext;
+    private final MDCWrapper mdcWrapper;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/dwp/requestcontext/RequestIdFilter.java:42]: Can we copy the unit test for this class from the claimant service?

File: api/src/main/resources/application.yml (hunk 1 above)
Comment (mikeSmith08) [reply api/src/main/resources/application.yml]: Done.

**Review by dwybourn — APPROVED** [review]

**Pull request merged**